docs/anonymity-configuration.rst

```diff
@@ -0,0 +1,521 @@
+``tcp = tor`` section causes Tahoe to use Tor when connecting to the
+Introducer, hiding it's IP address. The node will then announce itself to all
+clients using ``tub.location``, so clients will know that they must use Tor
+to reach this server (and not revealing it's IP address through the
+announcement). When clients connect to the onion address, their packets will
+flow through the anonymizing network and eventually land on the local Tor
+daemon, which will then make a connection to PORT on localhost, which is
+where Tahoe is listening for connections.
+
+Follow a similar process to build a Tahoe server that listens on I2P. The
+same process can be used to listen on both Tor and I2P (``tub.location =
+tor:ONION.onion:VIRTPORT,i2p:ADDR.i2p``). It can also listen on both Tor and
+plain TCP (use-case 2), with ``tub.port = tcp:PORT``, ``tub.location =
+tcp:HOST:PORT,tor:ONION.onion:VIRTPORT``, and ``anonymous = false`` (and omit
+the ``tcp = tor`` setting, as the address is already being broadcast through
+the location announcement).
+
+
+Server anonymity, automatic configuration
+-----------------------------------------
+
+(note: this is not yet implemented, see Tahoe tickets `#2490`_ and `#2773`_
+for progress)
+
+
+To configure a server node to listen on an anonymizing network, create the
+node with the ``--listen=tor`` option. This requires a Tor configuration that
+either launches a new Tor daemon, or has access to the Tor control port (and
+enough authority to create a new onion service).
+
+This option will set ``anonymous = true``, ``[connections] tcp = tor``. It
+will allocate the necessary ports, instruct Tor to create the onion service,
+obtain the ``.onion`` address, and populate ``tub.port`` and ``tub.location``
+correctly.
+
+
+Performance and security issues
+===============================
+
+If you are running a server which does not itself need to be
+anonymous, should you make it reachable via an anonymizing network or
+not? Or should you make it reachable *both* via an anonymizing network
+and as a publicly traceable TCP/IP server?
+
+There are several trade-offs effected by this decision.
+
+NAT/Firewall penetration
+------------------------
+
+Making a server be reachable via Tor or I2P makes it reachable even if there
+are NATs or firewalls preventing direct TCP/IP connections to the server.
+
+Anonymity
+---------
+
+Making a Tahoe-LAFS server accessible *only* via Tor or I2P can be used to
+guarantee that the Tahoe-LAFS clients use Tor or I2P to connect. This prevents
+misconfigured clients from accidentally de-anonymizing themselves by connecting
+to your server through the traceable Internet.
+
+Also, interaction, through Tor, with a Tor Hidden Service may be more
+protected from network traffic analysis than interaction, through Tor,
+with a publicly traceable TCP/IP server.
+
+**XXX is there a document maintained by Tor developers which substantiates or refutes this belief?
```

Comment:
certainly if the storage server made it's service available over a public ip as well as onion service then a simple confirmation attack could be used to deanonymize the onion service